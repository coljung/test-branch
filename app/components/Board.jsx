import React from 'react';
import PropTypes from 'prop-types';

export default class Board extends React.Component {
    render() {
        let boardTitle = null;
        if (this.props.btnInTitle) {
            boardTitle = (
                <div className='clearfix titleWithButton'>
                    <h2>{this.props.title}</h2>
                    {this.props.btnInTitle}
                </div>
            );
        } else {
            boardTitle = (<h2>{this.props.title}</h2>);
        }
        return (
            <div className='board' id={this.props.id}>
                {boardTitle}
                {this.props.children}
            </div>
        );
    }
}

Board.propTypes = {
    title: PropTypes.string.isRequired,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.element),
        PropTypes.element,
    ]).isRequired,
    id: PropTypes.string,
    btnInTitle: PropTypes.element,
};

Board.defaultProps = {
    btnInTitle: null,
    id: '',
};
