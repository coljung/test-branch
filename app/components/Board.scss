
.board{
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    font-size: 16px;
    font-weight: 400;
    min-height: 200px;
    overflow: hidden;
    width: auto;
    z-index: 1;
    position: relative;
    background: #fff;
    border-radius: 2px;
    box-sizing: border-box;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,.14),0 3px 1px -2px rgba(0,0,0,.2),0 1px 5px 0 rgba(0,0,0,.12);
    margin: 10px;
    padding:15px;
}