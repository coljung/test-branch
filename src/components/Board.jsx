import React from 'react';
import PropTypes from 'prop-types';

const Board = props => (
    <div className='board'>
        <h2>{props.title}</h2>
        {props.children}
    </div>
);

Board.propTypes = {
    title: PropTypes.string.isRequired,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.element),
        PropTypes.element,
    ]).isRequired,
};

export default Board;
