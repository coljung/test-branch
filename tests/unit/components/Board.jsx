import React from 'react';
import { shallow } from 'enzyme';
import Board from '../../../app/components/Board';

describe('Board', () => {
    it('Should display title', () => {
        const output = shallow(
            <Board title="test">
                <div />
            </Board>
        );

        expect(output.find('h2').text()).toEqual('test');
    });

    it('Should contain children', () => {
        const output = shallow(
            <Board title="test">
                <p>Test</p>
            </Board>
        );

        expect(output.find('p')).toHaveLength(1);
        expect(output.find('p').text()).toEqual('Test');
    });
});
