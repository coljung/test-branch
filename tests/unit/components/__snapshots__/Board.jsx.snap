// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render correctly 1`] = `
<div
  className="board"
  id=""
>
  <h2>
    fsdfsdf
  </h2>
  Hello Jest!
</div>
`;

exports[`Board should render correctly with btnInTitle 1`] = `
<div
  className="board"
  id=""
>
  <div
    className="clearfix titleWithButton"
  >
    <h2>
      Hello World!
    </h2>
    <Button
      ghost={false}
      loading={false}
      prefixCls="ant-btn"
    >
      Foo Bar
    </Button>
  </div>
  Hello Jest!
</div>
`;
